xtrack/line.py

```diff
@@ -61,6 +61,64 @@ def _next_name(prefix, names, name_format='{}{}'):
         i += 1
     return name_format.format(prefix, i)
 
+def _dicts_equal(dict1, dict2):
+    if not isinstance(dict1, dict) or not isinstance(dict2, dict):
+        raise ValueError
+    if set(dict1.keys()) != set(dict2.keys()):
+        return False
+    for key in dict1.keys():
+        if hasattr(dict1[key], '__iter__'):
+            if not hasattr(dict2[key], '__iter__'):
+                return False
+            elif not np.array_equal(dict1[key], dict2[key]):
+                return False
+        elif dict1[key] != dict2[key]:
+            return False
+    return True
+
+def _apertures_equal(ap1, ap2):
+    if not _is_aperture(ap1) or not _is_aperture(ap2):
+        raise ValueError(f"Element {ap1} or {ap2} not an aperture!")
+    if ap1.__class__ != ap2.__class__:
+        return False
+    ap1 = ap1.to_dict()
+    ap2 = ap2.to_dict()
+    return _dicts_equal(ap1, ap2)
+
+def _lines_equal(line1, line2):
```

Comment:
This is not used in xtrack. Where is it needed? Is it covered by tests?

Reply:
I use it in the collimation repository when auto-generating a line, by writing to file and importing again; both should match (this is to rule out write errors).